lighthouse-core/gather/connections/cri.js

```diff
@@ -66,6 +78,16 @@ class CriConnection extends Connection {
           reject(new Error(`Unable to fetch webSocketDebuggerUrl, status: ${response.statusCode}`));
         });
       });
+
+      request.setTimeout(CONNECT_TIMEOUT, _ => {
+        request.abort();
+
+        const err = new Error('Timeout waiting for initial Debugger Protocol connection.');
+        err.code = 'CRI_TIMEOUT';
```

Comment:
yeah. It's used a bunch by node, so I stole it from them, but we can move back to parsing `error.message` if you want